verification/certifyKey.go

```diff
@@ -133,6 +118,37 @@ func TestCertifyKeySimulation(d TestDPEInstance, c DPEClient, t *testing.T) {
 	testCertifyKey(d, c, t, true)
 }
 
+func TestCertifyKey_Csr(d TestDPEInstance, c DPEClient, t *testing.T) {
+	ctx := getInitialContextHandle(d, c, t, false)
+
+	profile, err := GetTransportProfile(d)
+	if err != nil {
+		t.Fatalf("Could not get profile: %v", err)
+	}
+	digestLen := profile.GetDigestSize()
+
+	flags := CertifyKeyFlags(0)
+	label := make([]byte, digestLen)
+
+	// Get DPE leaf certificate from CertifyKey
+	certifyKeyResp, err := c.CertifyKey(ctx, label, CertifyKeyCsr, flags)
+	if err != nil {
+		t.Fatalf("[FATAL]: Could not certify key: %v", err)
+	}
+
+	wrappedCSR, err := pkcs7.Parse(certifyKeyResp.Certificate)
+	if err != nil {
+		t.Fatalf("[FATAL]: Could not unmarshal CSR CMS message: %v", err)
+	}
+
+	csr, err := x509.ParseCertificateRequest(wrappedCSR.Content)
```

Comment:
Currently it looks like we are parsing the CMS, the CSR, and the extensions within the CRI in the CSR. Do you think it would be good to parse and validate some more fields? Probably overkill to do all of them but a few things I have in mind: 
- version in certification_request_info
- subject name in certification_request_info
- public key
- signature alg/signature value

Reply:
Done. One thing I found is that in SerialNumberAndIssuer, I think the Serial Number is supposed to be the issuer's serial number rather than the subject serial number.

I can't 100% tell from the spec, but otherwise the cert verification fails.

Reply:
Yeah I think you're right. Thanks for catching this.